web/src/engine/websites/Delitoon.ts

```diff
@@ -0,0 +1,174 @@
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const url = new URL('/api/balcony-api-v2/contents/search', this.URI);
+        url.search = new URLSearchParams({
+            searchText: '',
+            isCheckDevice: 'true',
+            isIncludeAdult: 'true',
+            contentsThumbnailType: 'MAIN'
+        }).toString();
+        const { data } = await FetchJSON<APIResult<APIManga[]>>(this.CreateRequest(url));
+        return data.map(element => new Manga(this, provider, element.alias, element.title.trim()));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        await this.GetToken();
+        const url = new URL(`/api/balcony-api-v2/contents/${manga.Identifier}`, this.URI);
+        url.searchParams.set('isNotLoginAdult', 'true');
+        const { data } = await FetchJSON<APIResult<APIManga>>(this.CreateRequest(url));
+        return data.episodes.map(element => {
+            let title = element.title.trim();
+            title += element.subTitle ? ' : ' + element.subTitle.trim() : '';
+            return new Chapter(this, manga, element.alias, title);
+        });
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        await this.GetToken();
+        const url = new URL(`/api/balcony-api-v2/contents/${chapter.Parent.Identifier}/${chapter.Identifier}`, this.URI);
+        url.searchParams.set('isNotLoginAdult', 'true');
+        const apiresult = await FetchJSON<APIResult<APIPages>>(this.CreateRequest(url));
+        if (apiresult.result == 'ERROR') {
+            throw new Exception(R.Plugin_Common_Chapter_UnavailableError);
+        }
+        return apiresult.data.images.map(element => new Page(this, chapter, new URL(element.imagePath)));
+
+    }
+
+    protected CreateRequest(url: URL): Request {
+        const headers: HeadersInit = {
+            Referer: this.URI.origin,
+            'x-balcony-id': this.BalconyID,
+            'x-balcony-timeZone': this.Timezone,
+            'X-platform': this.Platform,
+        };
+        if (this.Token) headers['authorization'] = ' Bearer ' + this.Token.accessToken.token;
+        return new Request(url, {
+            method: 'GET',
+            headers: headers
+        });
+    }
+
+    private CreatePostRequest(url: URL, body: string, contentType: string = 'application/json'): Request {
+        const headers: HeadersInit = {
+            Referer: this.URI.origin,
+            'x-balcony-id': this.BalconyID,
+            'x-balcony-timeZone': this.Timezone,
+            'X-platform': this.Platform,
+            'Content-type': contentType
+        };
+        if (this.Token) headers['authorization'] = ' Bearer ' + this.Token.accessToken.token;
+        return new Request(url, {
+            method: 'POST',
+            headers: headers,
+            body: body,
+        });
+    }
+
+    protected async GetToken() {
```

Comment:
Method is called `GetToken` but is void.
Use a more appropriate name for the method.